doc/game.md

```diff
@@ -1,18 +1,90 @@
-# Game Loop
+# BaseGame
 
-The Game Loop module is a simple abstraction over the game loop concept. Basically most games are built upon two methods:
+`BaseGame` is the most basic and most commonly used `Game` class in Flame.
 
-* The render method takes the canvas ready for drawing the current state of the game.
-* The update method receives the delta time in seconds since last update and allows you to move to the next state.
+The `BaseGame` class implements a `Component` based `Game` for you; basically it has a list of
+`Component`s and passes the `update` and `render` calls to all `Component`s that have been added to
+the game. You can still extend those methods to add custom behavior, and you will get a few other
+features for free, like the passing of `resize` methods (every time the screen is resized the
+information will be passed to the resize methods of all your components) and also a basic camera
+feature.
 
-The `Game` class can be extended and will provide these gameloop methods and then its instance Flutter widget tree via the `GameWidget`.
+We refer to this component based system as the Flame Component System, FCS for short.
```

Comment:
should we take this opportunity to mention it's loosely based on ECS but not quite the same, but that we do provide ecs via oxygen -> link to the docs. just to avoid confusion.

Reply:
I think @wolfenrain wanted to add that once Oxygen is more "done"?
Is FCS really loosely based on ECS? It kind of lacks all the core principles of ECS?